packages/grid/src/Grid.tsx

```diff
@@ -1314,26 +1342,47 @@ class Grid extends PureComponent<GridProps, GridState> {
       }
     }
 
-    this.setViewState({ top, left, topOffset, leftOffset });
+    this.setViewState(
+      { top, left, topOffset, leftOffset },
+      false,
+      stickyOptions
+    );
   }
 
   /**
    * Checks the `top` and `left` properties that are set and updates the isStuckToBottom/Right properties
    * Should be called when user interaction occurs
    * @param viewState New state properties to set.
    * @param forceUpdate Whether to force an update.
+   * @param stickyOptions Options for sticky behavior
    */
-  setViewState(viewState: Partial<GridState>, forceUpdate = false): void {
+
+  setViewState(
+    viewState: Partial<GridState>,
+    forceUpdate = false,
+    stickyOptions?: StickyOptions
+  ): void {
     if (!this.metrics) throw new Error('metrics not set');
 
     const { isStickyBottom, isStickyRight } = this.props;
     const { top, left } = viewState;
     const { lastTop, lastLeft } = this.metrics;
+
     if (top != null) {
-      this.setState({ isStuckToBottom: isStickyBottom && top >= lastTop });
+      this.setState({
+        isStuckToBottom:
+          isStickyBottom &&
+          top >= lastTop &&
+          (stickyOptions?.shouldStickBottom ?? false),
+      });
     }
     if (left != null) {
-      this.setState({ isStuckToRight: isStickyRight && left >= lastLeft });
+      this.setState({
+        isStuckToRight:
+          isStickyRight &&
+          left >= lastLeft &&
+          (stickyOptions?.shouldStickRight ?? false),
+      });
```

Comment:
Perhaps I named the sticky options incorrectly. I think we only want to update the options when sticky options are `true`, we shouldn't be updating `isStuckToRight` or `isStuckToBottom` at all if the values are false. So would be more like:

```
if (top != null && (stickOptions?.shouldStickToBottom ?? false)) {
  this.setState({ isStuckToBottom: isStickBottom && top >= lastTop });
}
```

As it is right now, you'll always wipe out the `isStuckToBottom`/`isStuckToRight` state whenever the cursor is moved, even if not a result of a user interaction.